cmd/gateway/b2/gateway-b2.go

```diff
@@ -595,14 +605,24 @@ func (l *b2Objects) DeleteObject(ctx context.Context, bucket string, object stri
 	if err != nil {
 		return err
 	}
-	reader, err := bkt.DownloadFileByName(l.ctx, object, 0, 1)
+
+	f, _, err := bkt.ListFileNames(l.ctx, 1, object, "", "")
 	if err != nil {
 		logger.LogIf(ctx, err)
 		return b2ToObjectError(err, bucket, object)
 	}
-	io.Copy(ioutil.Discard, reader)
-	reader.Close()
-	err = bkt.File(reader.ID, object).DeleteFileVersion(l.ctx)
+
+	if len(f) != 1 {
+		err := minio.ObjectNotFound{
+			Bucket: bucket,
+			Object: object,
+		}
+
+		logger.LogIf(ctx, err)
```

Comment:
I don't think we need to log this error, since it is totally normal to not find an object if the user enters a wrong file name.